### ISSUE TITLE
Add an 'onHover' event to the timeline

### ISSUE THREAD

**Issue by aaronpcz** [almende/vis#226]
Hi,

I'm very impressed with this library and have used it for 2 projects.

I've developed very easily a popup based on that you allowed HTML to be added in as content.

I'd like to know if you have any plans to make popups available to bind to hover over events.

Thanks,
Aaron

**Comment by AlexDM0** [comment]
Hi,

Over the last year a lot of feature requests have been made. We have just introduced our new website which has a list of the requested features. We have placed this request on that list. 

The list can be found here:
http://visjs.org/featureRequests.html

An explaination of the new system can be found here:
http://visjs.org/blog.html#New\ website\ for\ vis\.js\!

I would like to stress that this does not mean we abandon this request. Discussion here will continue if needed on this feature but we will close it to keep our Github issue page more of a bug-todo list.

Future feature requests will still be made here and then added to the website by us.

Regards,

Alex

**Comment by aaronpcz** [comment]
When creating the ItemData, you can include tool tip content wrapped within a div with the rest of the HTML content.   Including a class attribute named appropriately (example tooltip).

Using css this can just be a standard tooltip with the following css.

.tooltip {
    display: none;
    position: fixed;
}

.vis.timeline .item:hover .tooltip {
    display: block;
    z-index: 1235;
}

I checked out the project and I don't think this is worth a code change but you could include it in your documentation ?

**Comment by iphar** [comment]
aaronpcz, I tried your suggestion. it works perfectly at first sight but has two drawbacks:
1. page scrolling breaks the tooltip position. After I scroll my page (because timelines height is bigger than the page) the tooltips are positioned too low (offset from scrolling). This makes them unusable.
2. minor point: performance with fixed positioned tooltips makes Firefox Browser very slow.

Currently I don't have a solution, as I also didn't have success with proprietary libs like qtip.

**Comment by sujoy-roy** [comment]
I'm struggling to get any hover tooltips working. Would someone be able to provide an example of usage to put tooltip content into the dataset, please?

**Comment by VMannello** [comment]
Just wanted to add to this conversation. I needed truncated items on the timeline to expand on hover. I used this jQuery which could easily implement a Tool Tip, or anything else.
```javascript
var orgWidth;
var orgZ;
$(document).on("mouseenter mouseleave", ".vis-item", function(e) {
       if (e.type === "mouseenter") {
             orgWidth = $(this).width();
             orgZ = $(this).css('zIndex');
             $(this).css('zIndex',99999);
             if ($(this).width() < $(this).children().first().children().first().width()) {
                    $(this).width($(this).children().first().children().first().width() + 9);
             }
       } else {
            $(this).width(orgWidth);
            $(this).css('zIndex',orgZ);
            }
        });
```

**Comment by heptaman** [comment]
I'm using Angular.js and solved this by creating a global function setTooltip, where `selector` is the jQuery selector for the elements, `tooltipAttribute` is the property name you want to copy from the item, and `generations` is the number of parents above the item where you want to set the tooltip.

```javascript
function setTooltip(selector, tooltipAttribute, generations) {
    var items = angular.element(selector);
    angular.forEach(items, function(item, key) {
        var parent = item;
        for(var g = 0; g < generations; g++)
            parent = parent.parentElement;
        parent.title = item[tooltipAttribute];
    });
}
```

Then after creating the items, I call it like this:

```javascript
$timeout(function() {
    setTooltip('.vis-item-content', 'innerText', 2);
}, 0);
```

I am also using FullCalendar.io and need the exact same thing, so this function serves me twice:

```javascript
$timeout(function() {
    setTooltip('.fc-title', 'innerText', 2);
}, 0);
```

Hope it helps someone else... Enjoy! :)

**Comment by pdanielkoe** [comment]
I wanted to show the exact time when the mouse hover on top of the timeline ? is there any possible way to do it ? thanks in advanced

**Comment by mojoaxel** [comment]
Reopening as Feature-Request issue (see #2114). 
Everybody: Please feel free to implement this!

**Comment by mojoaxel** [comment]
@yotamberk fixed this with #2473
